packages/fullstack/src/adapters/axum_adapter.rs

```diff
@@ -369,15 +369,19 @@ fn apply_request_parts_to_response<B>(
     }
 }
 
-/// SSR renderer handler for Axum
-pub async fn render_handler<P: Clone + serde::Serialize + Send + Sync + 'static>(
-    State((cfg, ssr_state)): State<(ServeConfig<P>, SSRState)>,
+/// SSR renderer handler for Axum with added state.
+pub async fn render_handler_with_state<
+    P: Clone + serde::Serialize + Send + Sync + 'static,
+    S: Send + Sync + 'static,
```

Comment:
This limits the context that is injected to one state. Can you change this to a callback that modifies the `DioxusServerContext` to make it possible to insert as many states as you want?

Reply:
Sure, done. I think the APIs here might be nicer if we used a similar pattern to axum where there's only one state allowed, which can be composed of substates. In particular getting `register_server_fns_with_handler` to work is pretty hairy right now!

That can always happen in a future refactor though.